### PR TITLE
refactor(v-snackbar): converted to ts

### PR DESCRIPTION
**Pull request by nekosaur**
## Description
Converted v-snackbar to ts

## Motivation and Context
¯\\\_(ツ)\_/¯

## How Has This Been Tested?

visually, jest

## Markup:

<details>

```vue
<template>
  <v-app>

    <v-card>
      <v-card-text>
        <v-container fluid>
          <v-layout row wrap>
            <v-flex xs12 sm3>
              <v-checkbox
                v-model="x"
                label="Left"
                value="left"
              ></v-checkbox>
            </v-flex>

            <v-flex xs6 sm3>
              <v-checkbox
                v-model="x"
                label="Right"
                value="right"
              ></v-checkbox>
            </v-flex>

            <v-flex xs6 sm3>
              <v-checkbox
                v-model="y"
                label="Top"
                value="top"
              ></v-checkbox>
            </v-flex>

            <v-flex xs6 sm3>
              <v-checkbox
                v-model="y"
                label="Bottom"
                value="bottom"
              ></v-checkbox>
            </v-flex>

            <v-flex xs12 sm3>
              <v-checkbox
                v-model="mode"
                label="Multi-line (mobile)"
                value="multi-line"
              ></v-checkbox>
            </v-flex>

            <v-flex xs12 sm3>
              <v-checkbox
                v-model="mode"
                label="Vertical (mobile)"
                value="vertical"
              ></v-checkbox>
            </v-flex>

            <v-flex xs12 sm4 offset-sm4>
              <v-text-field
                v-model="text"
                label="Text"
                type="text"
              ></v-text-field>
            </v-flex>

            <v-flex xs12 sm4>
              <v-text-field
                v-model.number="timeout"
                label="Timeout"
                type="number"
              ></v-text-field>
            </v-flex>
          </v-layout>

        </v-container>
        <v-btn
          block
          color="primary"
          dark
          @click="snackbar = true"
        >
          Show Snackbar
        </v-btn>
      </v-card-text>

      <v-snackbar
        v-model="snackbar"
        :bottom="y === 'bottom'"
        :left="x === 'left'"
        :multi-line="mode === 'multi-line'"
        :right="x === 'right'"
        :timeout="timeout"
        :top="y === 'top'"
        :vertical="mode === 'vertical'"
      >
        {{ text }}
        <v-btn
          color="pink"
          flat
          @click="snackbar = false"
        >
          Close
        </v-btn>
      </v-snackbar>
    </v-card>
  </v-app>
</template>

<script>
  export default {
    data () {
      return {
        snackbar: false,
        y: 'top',
        x: null,
        mode: '',
        timeout: 6000,
        text: 'Hello, I\'m a snackbar'
      }
    }
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: src/components/VSnackbar/VSnackbar.ts [hunk 1]
```diff
@@ -4,12 +4,13 @@ import Colorable from '../../mixins/colorable'
 import Toggleable from '../../mixins/toggleable'
 import { factory as PositionableFactory } from '../../mixins/positionable'
 
+import mixins from '../../util/mixins'
+import { VNode } from '../../../node_modules/vue'
```
Comment (KaelWD) [inline src/components/VSnackbar/VSnackbar.ts]: Use the absolute path

File: src/components/VSnackbar/VSnackbar.ts [hunk 2]
```diff
@@ -4,12 +4,13 @@ import Colorable from '../../mixins/colorable'
...
+
 /* @vue/component */
-export default {
+export default mixins(Colorable, Toggleable, PositionableFactory(['absolute', 'top', 'bottom', 'left', 'right'])).extend({
```
Comment (KaelWD) [inline src/components/VSnackbar/VSnackbar.ts]: Split this line (remember the comment)

File: src/components/VSnackbar/VSnackbar.ts [hunk 3]
```diff
@@ -55,17 +56,17 @@ export default {
...
+        this.activeTimeout = window.setTimeout(() => {
           this.isActive = false
         }, this.timeout)
       }
     }
   },
 
-  render (h) {
+  render (h): VNode {
     const children = []
```
Comment (KaelWD) [inline src/components/VSnackbar/VSnackbar.ts]: `const children: VNodeArrayChildren = []`

Comment (nekosaur) [reply src/components/VSnackbar/VSnackbar.ts]: Do you mean `VNodeChildrenArrayContents`?

Comment (KaelWD) [reply src/components/VSnackbar/VSnackbar.ts]: Yeah something like that.

File: src/components/VSnackbar/VSnackbar.ts [hunk 4]
```diff
@@ -55,17 +60,17 @@ export default {
 
   methods: {
     setTimeout () {
-      clearTimeout(this.activeTimeout)
+      window.clearTimeout(this.activeTimeout)
```
Comment (KaelWD) [inline src/components/VSnackbar/VSnackbar.ts]: window shouldn't be needed for these afaik

Comment (nekosaur) [reply src/components/VSnackbar/VSnackbar.ts]: seemed like without it typescript was wanting to use the node versions, which returned `NodeJS.Timer`

Comment (KaelWD) [reply src/components/VSnackbar/VSnackbar.ts]: yeah weird - I get `number` from `setTimeout(() => 0)`, but `NodeJS.Timer` if I add the second argument.

**Pull request merged**